### PR TITLE
Firing open-close callbacks for managed components

### PR DESCRIPTION
**Pull request by VadymBezpalko**
## Description
Firing open-close callbacks for managed components

## Storybook
https://feature-tooltip-callbacks-improvement--613a8e945a5665003a05113b.chromatic.com/

## Checklist

**Obligatory:**

- [x] Self-review
- [ ] Unit & integration tests
- [ ] Storybook cases
- [ ] Design review
- [ ] Functional (QA) review

**Optional:**

- [ ] Accessibility cases (keyboard control, correct HTML markup, etc.)

### CODE REVIEW

**Review by MichalPaszowski — APPROVED** [review]

File: packages/react-components/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -91,7 +91,9 @@ export const Tooltip: React.FC<ITooltipProps> = (props) => {
     } else {
       visible && onClose?.();
```
Comment (MichalPaszowski) [inline packages/react-components/src/components/Tooltip/Tooltip.tsx:92]: shouldn't it fire regardless of the fact that tooltip is managed or not?

Comment (VadymBezpalko) [reply packages/react-components/src/components/Tooltip/Tooltip.tsx:92]: I've modified this component, so callback will be fired whenever `isVisible` value is changing

**Review by JCown — APPROVED** [review]
IMO this component gets a bit complicated

**Comment by VadymBezpalko** [comment]
> IMO this component gets a bit complicated

I agree. We will work on refactoring and improving Tooltip component 👌

**Comment by VadymBezpalko** [comment]
I created a task for the Tooltip refactor in our board #624

**Pull request merged**